crates/cext/src/transpiler/target.rs

```diff
@@ -0,0 +1,741 @@
+/// number of qubits
+///
+/// @param num_qubits The number of qubits the ``Target`` will explicitly support.
+///
+/// @return A pointer to the new ``Target``
+///
+/// # Example
+///     
+///     QkTarget *target = qk_target_new(5);
+///
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub extern "C" fn qk_target_new(num_qubits: usize) -> *mut Target {
+    let target = Target::new(
+        None,
+        Some(num_qubits),
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+    )
+    .unwrap();
+    Box::into_raw(Box::new(target))
+}
+
+/// @ingroup QkTarget
+/// Returns the number of qubits of this ``Target``.
+///
+/// @param target A pointer to the ``Target``.
+///
+/// @return The number of qubits this target can use.
+///
+/// # Example
+///     
+///     QkTarget *target = qk_target_new(5);
+///     uint32_t num_qubits = qk_target_num_qubits(target);
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_num_qubits(target: *const Target) -> usize {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe { const_ptr_as_ref(target) };
+    target.num_qubits.unwrap_or_default()
+}
+
+/// @ingroup QkTarget
+/// Returns the dt value of this ``Target``.
+///
+/// @param target A pointer to the ``Target``.
+///
+/// @return The dt value of this ``Target`` or `NaN` if not assigned.
+///
+/// # Example
+///     const dt = 10e-9;
+///     QkTarget *target = qk_target_new(5);
+///     qk_target_set_dt(target, 10e-9);
+///     double dt = qk_target_dt(target);
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_dt(target: *const Target) -> f64 {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe { const_ptr_as_ref(target) };
+    target.dt.unwrap_or(f64::NAN)
+}
+
+/// @ingroup QkTarget
+/// Returns the granularity value of this ``Target``.
+///
+/// @param target A pointer to the ``Target``.
+///
+/// @return The granularity value of this ``Target``.
+///
+/// # Example
+///     QkTarget *target = qk_target_new(5);
+///     // The value defaults to 1
+///     uint32_t granularity = qk_target_granularity(target);
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_granularity(target: *const Target) -> u32 {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe { const_ptr_as_ref(target) };
+    target.granularity
+}
+
+/// @ingroup QkTarget
+/// Returns the `min_length` value of this ``Target``.
```

Comment:
There are also some other places where this misses the second quote 🙂 
```suggestion
/// Returns the ``min_length`` value of this ``Target``.
```

Reply:
Addressed in 395a194